CodeMaid/Logic/Cleaning/FileHeaderLogic.cs

```diff
@@ -58,16 +61,72 @@ internal void UpdateFileHeader(TextDocument textDocument)
                 return;
             }
 
+            settingsFileHeader = ReplaceVariables(textDocument, settingsFileHeader);
+
             var cursor = textDocument.StartPoint.CreateEditPoint();
             var existingFileHeader = cursor.GetText(settingsFileHeader.Length);
 
-            if (!existingFileHeader.StartsWith(settingsFileHeader))
+            if (!existingFileHeader.StartsWith(textDocument.GetCodeLanguageComment()))
             {
                 cursor.Insert(settingsFileHeader);
                 cursor.Insert(Environment.NewLine);
             }
         }
 
+        /// <summary>
+        /// Replace the variables in the header if any
+        /// </summary>
+        /// <param name="textDocument"></param>
```

Comment:
Please finish filling out the XML documentation (e.g. settingsFileHeader should be included, textDocument should be defined).